ed25519.go

```diff
@@ -124,11 +134,18 @@ func Verify(publicKey *[PublicKeySize]byte, message []byte, sig *[SignatureSize]
 	edwards25519.ScReduce(&hReduced, &digest)
 
 	var R edwards25519.ProjectiveGroupElement
-	var b [32]byte
-	copy(b[:], sig[32:])
-	edwards25519.GeDoubleScalarMultVartime(&R, &hReduced, &A, &b)
+	var s [32]byte
+	copy(s[:], sig[32:])
+
+	// https://tools.ietf.org/html/rfc8032#section-5.1.7 requires that s be in
+	// the range [0, order) in order to prevent signature malleability.
+	if !edwards25519.ScMinimal(&s) {
```

Comment:
This also contains the diff for the upstream changes from x/crypto? I think we should merge this in a separate PR first and add the `VerifyUncompressedKey` here only (by rebasing / merging).

Reply:
Okay!

Reply:
You can keep the PR open. I'll merge #6 after making sure the diff exactly matches the changes on x/crypto. Then, you can rebase on the branch that contains the merge.